NMS-14619: Quick Start: Determine service availability (#5350)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/service-assurance/introduction.adoc

[[ga-service-assurance]]
= Service Assurance

This section describes how {page-component-title} tests the availability of a service or device and measures its latency.

> Refer to the xref:operation:quick-start/service-assurance.adoc[service assurance topic] in the Quick Start section for additional information on determining service availability.

Author: Since this isn't the exact name of the topic, I think we can stick with lower case here.

In {page-component-title} a service monitor framework fulfils these tasks.

> The main component, Pollerd, provides the following functionality:

Reviewer: ```suggestion
The main component, pollerd, provides the following functionality:
```

> The main component, Pollerd, provides the following functionality:

Reviewer 2: ```suggestion
The main component, pollerd, provides the following functionality:
```

> In {page-component-title}, a service monitor framework fulfils these tasks.

Author: LOL. There are a few like that. I forget what the other ones are at the moment.

* Tracks the status of a management resource or an application for availability calculations.
* Measures response times for service quality.
* Correlates node and interface outages based on a <<deep-dive/service-assurance/critical-service.adoc#ga-service-assurance-critical-service, critical service>>.

> The following image shows the model and representation of availability and response time.

Reviewer 2: ```suggestion
The following image shows the model and representation of availability and response time:
```

> .Representation of latency measurement and availability

Reviewer 2: Since the previous line describes the image, I don't think we need the alt text.

```suggestion
```

image::service-assurance/01_node-model.png[]

> This information is based on service monitors which Pollerd stores and runs.

Reviewer: ```suggestion
This information is based on service monitors which pollerd stores and runs.
```

> This information is based on service monitors which Pollerd stores and runs.

Reviewer 2: ```suggestion
This information is based on service monitors which pollerd stores and runs.
```

A service can have any arbitrary name and is associated with a service monitor.

> For example, we can define two services with the name `HTTP` and `HTTP-8080`, both are associated with the HTTP Service Monitor but use a different TCP port configuration parameter.

Reviewer 2: ```suggestion
For example, we can define two services with the name `HTTP` and `HTTP-8080`.
Both are associated with the HTTP Service Monitor, but each uses a different TCP port configuration parameter.
```

> The following figure shows how Pollerd interacts with other components in {page-component-title} and applications or agents to be monitored.

Reviewer: ```suggestion
The following figure shows how pollerd interacts with other components in {page-component-title} and applications or agents to be monitored.
```

> The following figure shows how Pollerd interacts with other components in {page-component-title} and applications or agents to be monitored.

Reviewer 2: ```suggestion
The following figure shows how pollerd interacts with other components in {page-component-title} and applications or agents to be monitored.
```

> The following figure shows how pollerd interacts with other components in {page-component-title} and applications or agents to be monitored.

Author: When I build it locally and look at the file in GitHub, the figure is right after the sentence.

> The following figure shows how pollerd interacts with other components in {page-component-title} and applications or agents to be monitored.

Reviewer 2: Huh, odd. It must be something on my side.

The availability is calculated over the last 24 hours and is shown in the Surveillance Views, SLA Categories, and the Node Detail pages.
Response times are displayed as resource graphs of the IP interface on the Node Detail page.

> Configuration parameters of the service monitor can be seen in the Service page by clicking on the service name on the Node Detail page.

Author: The whole service assurance section could be rewritten.

TIP: The Service page also includes timestamps that indicate the last time at which the service was polled and found to to be Up (last good) or down (last fail).

> These fields help to validate that Pollerd is polling the services as expected.

Reviewer: ```suggestion
These fields help to validate that pollerd is polling the services as expected.
```

> These fields help to validate that Pollerd is polling the services as expected.

Reviewer 2: ```suggestion
These fields help to validate that pollerd is polling the services as expected.
```

> When a service monitor detects an outage, Pollerd sends an event. which is used to create an alarm.

Reviewer: ```suggestion
When a service monitor detects an outage, pollerd sends an event. which is used to create an alarm.
```

> When a service monitor detects an outage, Pollerd sends an event. which is used to create an alarm.

Reviewer 2: ```suggestion
When a service monitor detects an outage, pollerd sends an event. which is used to create an alarm.
```

You can also use events to generate notifications for on-call network or server administrators.

> The following images show the interaction of Pollerd in {page-component-title}.

Reviewer: ```suggestion
The following images show the interaction of pollerd in {page-component-title}.
```

> The following images show the interaction of Pollerd in {page-component-title}.

Reviewer 2: ```suggestion
The following image shows the interaction of pollerd in {page-component-title}:
```

> .Service assurance with Pollerd in {page-component-title}

Reviewer: ```suggestion
.Service assurance with pollerd in {page-component-title}
```

> .Service assurance with Pollerd in {page-component-title}

Reviewer 2: Since the previous line describes the image, I don't think we need the alt text.

```suggestion
```
